arcgis-ios-sdk-samples/Route and directions/Navigate route/NavigateRouteViewController.swift

```diff
@@ -19,51 +19,57 @@ import ArcGIS
 // MARK: - Navigate route View Controller
 
 class NavigateRouteViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The label to display navigation status.
+    @IBOutlet var statusLabel: UILabel!
+    /// The button to start navigation.
+    @IBOutlet var navigateBarButtonItem: UIBarButtonItem!
+    /// The button to reset navigation.
+    @IBOutlet var resetBarButtonItem: UIBarButtonItem!
+    /// The button to recenter the map to navigation pan mode.
+    @IBOutlet var recenterBarButtonItem: UIBarButtonItem!
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemap: .navigationVector())
+            let stopSymbol = AGSSimpleMarkerSymbol(style: .diamond, color: .orange, size: 20)
+            let stopGraphics = makeStops().map { AGSGraphic(geometry: $0.geometry, symbol: stopSymbol) }
+            let routeGraphics = [routeAheadGraphic, routeTraveledGraphic]
```

Comment:
Could all of this go into `makeRouteOverlay()`? Then the last line of this property observer could just be `mapView.graphicsOverlays.add(makeRouteOverlay())`.

Reply:
Yes sure. Recently I'm trying with better coding style and had a question: **when should I use instance properties in a function?**

My current understanding is to only access instance properties in a function that does not take arguments and/or does not return. e.g. `func loadGeodatabase()` or `func makeMap()` or `func someAsync(completion: @escaping () -> Void)`.

Otherwise I tend to create a "pure function" that does not deal with instance properties. (to achieve that, sometimes I pass instance property as an argument to a pure function).

Is my understanding correct? Also @vquach2404 as your knowledge in programming should be deeper than mine.

Much appreciated! 😄

Reply:
I tried to find some information in the [Swift doc](https://docs.swift.org/swift-book/LanguageGuide/Properties.html#:~:text=In%20addition%20to%20properties%2C%20you,property%20is%20not%20accessed%20directly) but it didn't really provide much of an answer. I would suggest reaching out to Mikey from BNR in addition to taking Phil's advice. 🙂

Reply:
> I would suggest reaching out to Mikey from BNR in addition to taking Phil's advice. 

Yes that is a good idea! Thanks for the heads up.

Reply:
There is no such rule that I know of for when to make a function pure and when not to. Pure functions are nice in some ways, but can lead to less straightforward code, which isn't necessarily good for sample code where the most important thing is to show users how they can use the ArcIGS runtime in their apps.